### PR TITLE
[backport] Support SciPy 1.9

### PR DESCRIPTION
**Pull request by kmaehashi**
Backport of #6962 by @takagi

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -4,4 +4,4 @@ sphinx-copybutton==0.5.0
 
 Cython==0.29.*
 numpy==1.23.*
-scipy==1.8.*
+scipy==1.9.*
```
Comment (kmaehashi) [inline docs/requirements.txt:7]:
```suggestion
scipy==1.8.*
```

File: docs/source/install.rst [hunk 2]
```diff
@@ -23,11 +23,11 @@ Requirements
 Python Dependencies
 ~~~~~~~~~~~~~~~~~~~
 
-NumPy/SciPy-compatible API in CuPy v11 is based on NumPy 1.23 and SciPy 1.8, and has been tested against the following versions:
+NumPy/SciPy-compatible API in CuPy v12 is based on NumPy 1.23 and SciPy 1.9, and has been tested against the following versions:
```
Comment (kmaehashi) [inline docs/source/install.rst:26]: Adjust docs for v11 backport

```suggestion
NumPy/SciPy-compatible API in CuPy v11 is based on NumPy 1.23 and SciPy 1.8, and has been tested against the following versions:
```

**Review by takagi — DISMISSED** [review]
LGTM!

**Pull request merged**